targets/TARGET_STM/i2c_api.c

```diff
@@ -514,7 +525,7 @@ int i2c_byte_read(i2c_t *obj, int last) {
     timeout = FLAG_TIMEOUT;
     while (__HAL_I2C_GET_FLAG(handle, I2C_FLAG_RXNE) == RESET) {
         if ((timeout--) == 0) {
-            return -1;
+            return 2;
```

Comment:
Can you clarify the reason for this change? Seems like this should return an error instead of what could be considered valid data?